sonic-net/sonic-buildimage
# Add BGPMonitorV6 session

#### Why I did it
For route registry service, IBGP session needs to be set up from BGP sentinel service (host) to SONiC. Add BGPMonV6 session in order to avoid passive listen range conflict with existing BGPS in device rollout. 
Add BGPMonV6 peer-group in monitor template and using BGPMonV6 route-maps for this new peer group. This change would not affect existing bgp monitor v4 session.

##### Work item tracking
- Microsoft ADO **(number only)**: 24610851

#### How I did it
Add BGPMON_V6 peer-group and related peer-group configuration.
Add route-map FROM_BGPMON_V6 and TO_BGPMON_V6. For FROM_BGPMON_V6, if match special community, accept otherwise deny all.
If BGP monitor address is V6, add to BGPMON_V6 peer-group.
Add BGP monitor V6 related UT.

#### How to verify it
Testcases and UT newly added would pass.
Setup IPv6 BGPMonitor session in minigraph, and load minigraph, show CONFIG_DB and "show runningconfig bgp", configuration would be loaded successfully.
Using t1-lag topo and setup IBGP session from ptf to SONiC loopback address, IBGP session would up.
Advertise route from ptf to T1 with special_community, higher local-preference and no-export community. In T1, show bgp route, the result is "Not advertise to any EBGP peer".
Withdraw the route in ptf, in T1, route would advertise to EBGP peers.
Advertise route from T1 that does not match special_community, in T1, would not see the route in show bgp route.



#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
